src/components/MediaUpload.tsx

```diff
@@ -90,37 +92,58 @@ function MediaUpload(props: IProps) {
     if (uploadMethod && files) {
       uploadMethod(files);
     }
-    setFiles(undefined);
-    setName('');
+    handleFiles(undefined);
     setOpen(false);
   };
   const handleCancel = () => {
     if (cancelMethod) {
-      setFiles(undefined);
-      setName('');
+      handleFiles(undefined);
       cancelMethod();
     }
     setOpen(false);
   };
-  const fileName = (files: FileList) => {
+  const fileName = (files: File[]) => {
     return files.length === 1
       ? files[0].name
       : files.length.toString() + ' files selected';
   };
 
+  const handleFiles = (files: FileList | undefined) => {
+    if (files) {
+      var goodFiles = Array.from(files).filter((s) =>
+        acceptextension[uploadType].includes(
+          (path.extname(s.name) || '.xxx').substring(1)
+        )
+      );
+      if (goodFiles.length < files.length) {
+        var rejectedFiles = Array.from(files).filter(
+          (s) =>
+            !acceptextension[uploadType].includes(
+              (path.extname(s.name) || '.xxx').substring(1)
+            )
+        );
+        showMessage(
+          t.invalidFile + rejectedFiles.map((f) => f.name).join(', ')
```

Comment:
My only comment is here.  It's probably better for localization to use a substitution in the message rather than appending the list to the end. Some languages may need text after the list as well as before.